hazelcast-hibernate5/src/test/java/com/hazelcast/hibernate/PhoneHomeTest.java

```diff
@@ -0,0 +1,77 @@
+package com.hazelcast.hibernate;
+
+import org.hibernate.Hibernate;
+import org.hibernate.boot.spi.SessionFactoryOptions;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.Arrays;
+import java.util.Map;
+import java.util.Properties;
+
+import static java.util.stream.Collectors.toMap;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@RunWith(MockitoJUnitRunner.class)
+public class PhoneHomeTest {
+
+    @Mock
+    PhoneHomeService phoneHomeService;
+
+    @Mock
+    SessionFactoryOptions sessionFactoryOptions;
+
+    @Test
+    public void localRegionRegistryTest() {
+        HazelcastLocalCacheRegionFactory regionFactory = new HazelcastLocalCacheRegionFactory();
+        regionFactory.setPhoneHomeService(phoneHomeService);
+        regionFactory.start(sessionFactoryOptions, new Properties());
+        verify(phoneHomeService, times(1)).start();
+        regionFactory.stop();
+        verify(phoneHomeService, times(1)).shutdown();
+    }
+
+    @Test
+    public void distributedRegionRegistryTest() {
+        HazelcastCacheRegionFactory regionFactory = new HazelcastCacheRegionFactory();
+        regionFactory.setPhoneHomeService(phoneHomeService);
+        regionFactory.start(sessionFactoryOptions, new Properties());
+        verify(phoneHomeService, times(1)).start();
+        regionFactory.stop();
+        verify(phoneHomeService, times(1)).shutdown();
+    }
```

Comment:
These tests ensure only if particular methods are called and I think we'd be safer if we additionally tested if a GET request is fired against a supplied URL.